src/rez/cli/test.py

```diff
@@ -67,3 +95,75 @@ def command(opts, parser, extra_arg_groups=None):
 
         if returncode:
             sys.exit(returncode)
+
+
+def is_dev_run(name):
+    """
+    Decides if is a dev_run, based on the package name
+    Args:
+        name: name of package
+
+    Returns: True or False
+    """
+    return not name or name == '.'
+
+
+def get_package(path):
+    """
+    Function validates pkg name and indicates if tests were run in developer mode
+    Args:
+        path: path to the package
+
+    Returns: Tuple of values (package_name, is_dev_run)
+    """
+    try:
+        return get_developer_package(path)
+    except PackageMetadataError:
+        print("Not a valid rez package. Make sure you are in package's root directory", file=sys.stderr)
+        return None
+
+
+def prepare_dev_env_package(package):
+    """
+    Prepares a dev environment for tests. If a package is not installed in the local packages path
+    or the tests on the package[py/yaml] are outdated, it installs the package
+    or updates the package file with the new test targets.
+    Args:
+        package: package to be built
+    """
+    try:
+        local_package_path = os.path.join(config.local_packages_path, package.name, str(package.version))
+        local_package = get_developer_package(local_package_path)
+    except PackageMetadataError:
+        install_package_in_local_packages_path()
+    else:
+        update_package_in_local_packages_path(package, local_package)
+
+
+def install_package_in_local_packages_path():
+    """
+    Installs the package in the local_packages_path (defined in config.local_packages_path)
+    """
+    original_args = sys.argv
+    sys.argv = [original_args[0]] + ['-i']
+    try:
+        run("build")
+    except SystemExit as exit_code:
+        if exit_code.code:
+            raise
+    sys.argv = original_args
+
+
+def update_package_in_local_packages_path(package, installed_package):
```

Comment:
Again this feels arbitrary - the package is being kept up-to-date wrt its tests, but not its actual payload (ie if source was changed). I think we should just leave it up to the developer to reinstall the package.

Having said that, something that I think _would_ be acceptable in developer mode is if the test definitions are always taken from the developer package, rather than from the installed package.

Reply:
In that case, if they are taken from the developer package, I think people will be happy